src/main/lua/exasolrls/adapter_properties.lua

```diff
@@ -0,0 +1,122 @@
+---
+-- This class provides an abstraction to the access of adapter properties.
+--
+local text = require("text")
+
+local M = {
+    SCHEMA_NAME_PROPERTY = "SCHEMA_NAME",
+    TABLE_FILTER_PROPERTY = "TABLE_FILTER",
+    raw_properties = {}
+}
+
+---
+-- Create a new instance of adapter properties
+--
+-- @param object preinitialized object (optional)
+-- 
+-- @return new instance
+-- 
+function M:new (object)
+    object = object or {}
+    self.__index = self
+    setmetatable(object, self)
+    return object
+end
+
+---
+-- Get the value of a property.
+-- 
+-- @param property_name name of the property to get
+--
+-- @return property value
+-- 
+function M:get(property_name)
+    return self.raw_properties[property_name]
+end
+
+---
+-- Check if the property is set.
+-- 
+-- @param property_name name of the property to check
+-- 
+-- @return <code>true</code> if the property is set (i.e. not <code>nil</code>)
+-- 
+function M:is_property_set(property_name)
+    return self:get(property_name) ~= nil
+end
+
+---
+-- Check if the property has a non-empty value.
+-- 
+-- @param property_name name of the property to check
+-- 
+-- @return <code>true</code> if the property has a non-empty value (i.e. not <code>nil</code> or an empty string)
+--
+function M:has_value(property_name)
+    local value = self:get(property_name)
+    return value ~= nil and value ~= ""
+end
+
+---
+-- Check if the property value is empty.
+-- 
+-- @param property_name name of the property to check
+-- 
+-- @return <code>true</code> if the property's value is empty (i.e. the property is set to an empty string)
+--
+function M:is_empty(property_name)
+    return self:get(property_name) == ""
+end
+
+---
+-- Validate the adapter properties.
+-- 
+-- @raise validation error
+-- 
+-- @return self for fluent programming
+--
+function M:validate()
+    if not self:has_value(M.SCHEMA_NAME_PROPERTY) then
+        error('F-RLS-PROP-1: Missing mandatory property "SCHEMA_NAME". Please define the name of the source schema.');
```

Comment:
You could also use the constant `M.TABLE_FILTER_PROPERTY` here in the error message like below.
```suggestion
        error('F-RLS-PROP-1: Missing mandatory property "' .. M.TABLE_FILTER_PROPERTY .. '". Please define the name of the source schema.');
```